app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -154,6 +156,30 @@ class EventDetailsFragment : Fragment() {
             loadTicketFragment()
         }
 
+        eventViewModel.loadOrders()
+        eventViewModel.orders
```

Comment:
You can check if eventViewModel.orders.value == null before calling eventViewModel.loadOrders(). If it is not null, then you can call the method below immediately without making the network call again. This should prevent making the API call again after screen rotation or coming back to this fragment from backstack. 
```
it.forEach { order ->
                    if (order.event?.id == safeArgs.eventId) {
                        rootView.alreadyRegisteredLayout.visibility = View.VISIBLE
                        orderId = order.id
                        rootView.alreadyRegisteredLayout.setOnClickListener {
                            order.identifier?.let { identifier ->
                                EventDetailsFragmentDirections.actionEventDetailsToOrderDetail(
                                    eventId = safeArgs.eventId,
                                    orderId = order.id,
                                    orderIdentifier = identifier
                                )
                            }?.let { navigation ->
                                findNavController(rootView).navigate(navigation)
                            }
                        }
                        return@forEach
                    }
                }
```